ui5/browser/controller/Browser.controller.js

```diff
@@ -142,54 +163,191 @@ sap.ui.define(['sap/ui/core/Component',
+         this.globalId++;
+
+         oTabContainerItem.addContent(oCodeEditorFragment);
+         oTabContainer.addItem(oTabContainerItem);
+         oTabContainer.setSelectedItem(oTabContainerItem);
+
+         sap.ui.getCore().byId("CodeEditor" + ID + "Editor").setModel(new JSONModel({
             code: "",
             ext: "",
             filename: "",
             fullpath: "",
             modified: false
          }));
+      },
 
-         let splitterUpperContent = this.getView().byId("defaultCodeEditor").getContent()[0].mAggregations.contentAreas[0].getContent();
-         splitterUpperContent[0].attachChange(this.onChangeFile, this);
-         splitterUpperContent[1].attachPress(this.onSaveAs, this);
-         splitterUpperContent[2].attachPress(this.onSaveFile, this);
-         splitterUpperContent[3].attachPress(this.onRunMacro, this);
-
-         defaultCodeEditor.attachChange( function() {
+      newCodeEditorFragment: function (ID) {
+         let fragment = new Splitter("CodeEditor" + ID + "Splitter", {
+            orientation: "Vertical",
+            contentAreas: [
+               new Toolbar("CodeEditor" + ID + "Toolbar", {
+                  content: [
+                     new FileUploader("CodeEditor" + ID + "FileUploader", {}),
+                     new Button("CodeEditor" + ID + "SaveAs", {
+                        text: "Save as...",
+                        tooltip: "Save current file as...",
+                     }),
+                     new Button("CodeEditor" + ID + "Save", {
+                        text: "Save",
+                        tooltip: "Save current file",
+                     }),
+                     new Button("CodeEditor" + ID + "Run", {
+                        text: "Run",
+                        tooltip: "Run Current Macro",
+                        icon: "sap-icon://play",
+                        enabled: false,
+                     }),
+                  ],
+                  layoutData: new SplitterLayoutData("CodeEditor" + ID + "SplitterLayoutData", {
+                     size: "35px",
+                     resizable: false
+                  })
+               }),
+               new CodeEditor("CodeEditor" + ID + "Editor", {
+                  height: "100%",
+                  colorTheme: "default",
+                  type: "c_cpp",
+                  value: "{/code}"
+               })
+            ]
+         });
+         sap.ui.getCore().byId("CodeEditor" + ID + "FileUploader").attachChange(this.onChangeFile, this);
+         sap.ui.getCore().byId("CodeEditor" + ID + "SaveAs").attachPress(this.onSaveAs, this);
+         sap.ui.getCore().byId("CodeEditor" + ID + "Save").attachPress(this.onSaveFile, this);
+         sap.ui.getCore().byId("CodeEditor" + ID + "Run").attachPress(this.onRunMacro, this);
+         sap.ui.getCore().byId("CodeEditor" + ID + "Editor").attachChange(function () {
             this.getModel().setProperty("/modified", true);
```

Comment:
Is it working? That value is `this`?

Reply:
Yes, the this, doesn't refer to the controller this, but to the this of the code editor
So as i had the model like so :
```
sap.ui.getCore().byId("CodeEditor" + ID + "Editor").setModel(new JSONModel({
            code: "",
            ext: "",
            filename: "",
            fullpath: "",
            modified: false
         }));
```
the this of the event is the current object, so it can take the correct model
But with my new modifications i am going to make this better again

Reply:
For some reason i can use Model or model here :
```
new CodeEditor("CodeEditor" + ID + "Editor", {
                  height: "100%",
                  colorTheme: "default",
                  type: "c_cpp",
                  value: "{/code}",
                  change: function () {
                     this.getModel().setProperty("/modified", true);
                  }
               })
```
So I used your solution, to do the function right after the declaration

Reply:
If `this.GetModel()` return proper value - just use it.